Userland/Libraries/LibWeb/Fetch/Fetching/Fetching.cpp

```diff
@@ -1252,6 +1252,114 @@ WebIDL::ExceptionOr<JS::GCPtr<PendingResponse>> http_redirect_fetch(JS::Realm& r
     return main_fetch(realm, fetch_params, recursive);
 }
 
+// https://fetch.spec.whatwg.org/#network-partition-key
+struct NetworkPartitionKey {
+    HTML::Origin top_level_origin;
+    // FIXME: See https://github.com/whatwg/fetch/issues/1035
+    void* second_key = nullptr;
+};
+
+struct CachedResponse {
+    struct Header {
+        ByteBuffer name;
+        ByteBuffer value;
+    };
```

Comment:
Can we de-dup this and use `Web::Fetch::Infrastructure::Header`?

Reply:
Ah yep, can do. I was trying to avoid using any of the things such as HeaderList that are GC-allocated since we want to eventually have this caching business go over IPC in some way to be centrally managed.